### PR TITLE
MSWC FSCIL

### PR DESCRIPTION
**Pull request by V0XNIHILI**
Adding MSWC FSCIL functionality + example!

### CODE REVIEW

File: neurobench/data/datasets/generation.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/data/datasets/generation.py:1]: Yes sorry forgot to delete it

File: neurobench/benchmarks/benchmark.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/benchmarks/benchmark.py:1]: Cool thanks

File: neurobench/examples/mswc_fscil.py [hunk 1]
```diff
@@ -1,44 +1,82 @@
...
+
+### Evaluation phase ###
+dummy_train = lambda net, data: net
+
+# Get Datasets: evaluation + all test samples from base classes to test forgetting
+eval_set = MSWC(root=ROOT, subset="evaluation")
+base_test_set = MSWC(root=ROOT, subset="base", procedure="testing")
+
+# FewShot Dataloader used in incremental mode to generate class-incremental sessions
+few_shot_dataloader = FewShot(eval_set, n_way=10, k_shot=5, query_shots=100,
```
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/examples/mswc_fscil.py:32]: Yes sorry I forgot to mention that, I also had to make this change to torch_mate on my side but I don't have push rights so couldn't do it. 
Do you have time to quickly add the import in the init function @V0XNIHILI ? :)

File: neurobench/models/__init__.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/models/__init__.py:1]: https://github.com/NeuroBench/neurobench/pull/1/commits/a62e5a23156f258dbdf641a6d199a84dc8e8b31a

File: neurobench/models/M5.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/models/M5.py:1]: https://github.com/NeuroBench/neurobench/pull/1/commits/a62e5a23156f258dbdf641a6d199a84dc8e8b31a

File: neurobench/examples/mswc_fscil.py [hunk 2]
```diff
@@ -1,44 +1,82 @@
 import torch
 
-from speech2spikes import S2S
+import torchvision
+from torch.utils.data import DataLoader, ConcatDataset
+import torchaudio
+
+from torch_mate.data.utils import FewShot, IncrementalFewShot
```
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/examples/mswc_fscil.py]: Sorry forgot about it, this is deprecated. 
I just cleaned it: https://github.com/NeuroBench/neurobench/pull/1/commits/f0d9a729ca7ddb6406fd84d9336b815a3645a7a8

File: neurobench/datasets/MSWC.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/datasets/MSWC.py:1]: I added the detailed documentation to the MSWC class: https://github.com/NeuroBench/neurobench/pull/1/commits/a74a7c64bbedb9accf438c90b9749bad68ee6882
But also I am wondering if the download procedure should be as a function description or as it's own README file or something. What do you think?

File: neurobench/examples/mswc_fscil.py
[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/examples/mswc_fscil.py:1]: Hopefully the runtime error should be solved with this https://github.com/NeuroBench/neurobench/pull/1/commits/f0d9a729ca7ddb6406fd84d9336b815a3645a7a8 as we were using 8 workers for the dataloader by default and now I put it back to 1 and it also works on a small CPU for me.
Let me know if you still have the error and if you can show it to me.

[the remark this answers is not in the document]
Comment (Maxtimer97) [reply neurobench/examples/mswc_fscil.py:1]: For torch_mate dependencies I am not sure. We would just need to copy 2 files/3 classes from torch_mate from what we are using right now but I think it's up to @V0XNIHILI to say how he wants to proceed with it. 
If you are good with me copying the classes from torch_mate I can do it @V0XNIHILI .

**Comment by Maxtimer97** [comment]
https://github.com/NeuroBench/neurobench/blob/f0d9a729ca7ddb6406fd84d9336b815a3645a7a8/neurobench/datasets/MSWC.py#L15-L216

Should we move this list to another python file?

**Comment by Maxtimer97** [comment]
> Having some difficulties running the example, also question about the torch_mate dependency in the example

Hopefuly this should be now fixed with the latest commit : https://github.com/NeuroBench/neurobench/blob/2ef814c79e8d5d269b1acd4ba3d1f2cbdacf9631/neurobench/examples/mswc_fscil.py#L20

At least, it works on my side with the poetry install on a Windows system.

Basically, I just added the if main clause that maintains correctness for multiprocesses forking on Windows. 
Let me know if it also solves the problem for you.

**Comment by jasonlyik** [comment]
@Maxtimer97 It is running now! Small comments:
- I believe there should be a benchmark run at line 39 using the base_test_set in order to evaluate base accuracy.
- Line 90 torch.mean() is applied to a list, not tensor, so throwing an error for me.
- Total runtime of the script on my M1 macbook is ~14 min. I think data loading optimization can be considered but this doesn't seem prohibitively slow!

**Comment by Maxtimer97** [comment]
> @Maxtimer97 It is running now! Small comments:
> 
> * I believe there should be a benchmark run at line 39 using the base_test_set in order to evaluate base accuracy.
> * Line 90 torch.mean() is applied to a list, not tensor, so throwing an error for me.
> * Total runtime of the script on my M1 macbook is ~14 min. I think data loading optimization can be considered but this doesn't seem prohibitively slow!

Thanks! First and Second comments should be solved with this last commit: https://github.com/NeuroBench/neurobench/pull/1/commits/6f2cd8c4f38198d630b8405618ca36ac1cd80e9e

I will be exploring the data loading further now.

**Comment by jasonlyik** [comment]
Closing branch, other mswc task branches more updated.
